### PR TITLE
Simplify tests

### LINKED ISSUE #82: Use seed() in tests?
I'd like to increase the Provider test coverage a bit. To do so I checked out the existing tests and saw different approaches taken:

some tests use seed() to ensure a certain result is returned, e.g.
https://github.com/FakerPHP/Faker/blob/907b307d0d40034d64aa1377311169238943c68a/test/Faker/Provider/uk_UA/PersonTest.php#L11-L18

other tests just check that a non-empty result is returned, e.g.
https://github.com/FakerPHP/Faker/blob/907b307d0d40034d64aa1377311169238943c68a/test/Faker/Provider/bn_BD/PersonTest.php#L18-L22

Another approach would be to test that we're dealing with a valid string (in the case of `firstNameMale()`) or via regex a valid pattern.

Is there another solution I have overlooked/any other ideas? Which solution do you prefer?

### PR DESCRIPTION
**Pull request by IonBazan**
This PR removes a lot of duplicated code and refactors all the tests by:

- [x] Initializing `Generator` instance for each test in `setUp` method
- [x] Sets the seed to `1` to make results predictable
- [x] Adding `protected function getProviders(): iterable` allowing developers to easily specify providers needed in the test
- [x] Migrated most of `$this->assert*` and `$this->fail()` to static calls
- [x] Fixes #82 
- [x] Fixes styles using `@Symfony` rule

### CODE REVIEW

File: test/Faker/Provider/en_US/PhoneNumberTest.php [hunk 1]
```diff
@@ -2,84 +2,75 @@
...
-    {
-        $faker = new Generator();
-        $faker->addProvider(new PhoneNumber($faker));
-        $this->faker = $faker;
-    }
-
     public function testPhoneNumber()
     {
-        for ($i = 0; $i < 100; $i++) {
+        for ($i = 0; $i < 100; ++$i) {
```
Comment (krsriq) [inline test/Faker/Provider/en_US/PhoneNumberTest.php:12]: We could remove all of these for loops that test different randomized versions when we're seeding the number generator I guess?

File: test/Faker/Provider/ProviderOverrideTest.php [hunk 2]
```diff
@@ -6,13 +6,7 @@
...
 /**
- * Class ProviderOverrideTest
- *
- * @package Faker\Test\Provider
- *
- * This class tests a large portion of all locale specific providers. It does not test the entire stack, because each
- * locale specific provider (can) has specific implementations. The goal of this test is to test the common denominator
- * and to try to catch possible invalid multi-byte sequences.
+ * Class ProviderOverrideTest.
  */
```
Comment (bramceulemans) [inline test/Faker/Provider/ProviderOverrideTest.php:10]: Why remove the explanation?

Just repeating the class name doesn't add much value. This comment might as well be removed completely.

File: test/Faker/Provider/en_ZA/PhoneNumberTest.php [hunk 3]
```diff
@@ -25,26 +25,26 @@ public function testPhoneNumber()
...
             $digits = array_values(array_filter(str_split($number), 'ctype_digit'));
 
-            if (count($digits) === 11) {
+            if (11 === count($digits)) {
                 self::assertEquals('0', $digits[0]);
             }
 
-            $areaCode = $digits[0] . $digits[1] . $digits[2] . $digits[3];
-            self::assertContains($areaCode, array('0800', '0860', '0861', '0862'));
+            $areaCode = $digits[0].$digits[1].$digits[2].$digits[3];
```
Comment (bramceulemans) [inline test/Faker/Provider/en_ZA/PhoneNumberTest.php:36]: Personally I don't like this, is this really in our style guide?

File: test/Faker/Provider/es_VE/CompanyTest.php [hunk 4]
```diff
@@ -8,7 +8,7 @@
 final class CompanyTest extends TestCase
 {
     /**
-     * national Id format validator
+     * national Id format validator.
```
Comment (bramceulemans) [inline test/Faker/Provider/es_VE/CompanyTest.php:11]: Capitalize this please.

**Review by bramceulemans — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Please revert all code style changes, and send them as a separate PR. The current code style decision was to use PSR-12, enforced by StyleCI, as per the contribution guidelines. Don't run external code style fixing tools.

File: test/Faker/Provider/cs_CZ/PersonTest.php [hunk 5]
```diff
@@ -23,16 +23,20 @@ public function testBirthNumber()
...
+            }
+            if ($year >= 2004 && $month > 20) {
+                $month -= 20;
+            }
 
             self::assertTrue(checkdate($month, $day, $year), "Birth number $birthNumber: date $year/$month/$day is invalid.");
 
             // check CRC if presented
-            if (strlen($birthNumber) == 10) {
+            if (10 == strlen($birthNumber)) {
```
[the remark this answers is not in the document]
Comment (bramceulemans) [reply test/Faker/Provider/cs_CZ/PersonTest.php:36]: I can see one advantage here, you don't have to scan to the end of the (maybe very long) statement, to see the number.

[the remark this answers is not in the document]
Comment (mfn) [reply test/Faker/Provider/cs_CZ/PersonTest.php:36]: meta discussion/idea: can either style be enforced with the style fixer?

Then there's no discussion anymore :)

**Comment by GrahamCampbell** [comment]
Please re-send this without any code style changes. If you want to change the code style, please open a separate PR and do it only using StyleCI.

**Comment by GrahamCampbell** [comment]
> Migrated most of $this->assert* and $this->fail() to static calls

This can be done using StyleCI.

**Pull request closed without merging**